### PR TITLE
[NO MERGE] Deferred err handling

### PR DESCRIPTION
**Pull request by derwolfe**
This is for discussion only.

### CODE REVIEW

File: src/cloudpassage_lib/core.clj [hunk 1]
```diff
@@ -44,23 +45,71 @@
...
+      (md/chain
+       (invoker)
+       (fn [val]
+         (let [errval? (instance? Exception val)]
+           (cond
+             ;; stop trying
+             (and (= tries stop) errval?)
+             (do
+               (error "Failed retrying" tries "times; stopping")
+               (Exception. "Failed retrying; stopping."))
```
Comment (lvh) [inline src/cloudpassage_lib/core.clj]: Hm. I thought you were supposed to raise, and that the deferred machinery catches it; but I may be confusing with Twisted :)

Comment (lvh) [inline src/cloudpassage_lib/core.clj]: e.g. for chain, this is true: you either return a deferred that is in an error state, or you raise.

Comment (lvh) [inline src/cloudpassage_lib/core.clj]: Oh, hm. Reading the documentation, that looks like it might be chain-specific behavior. Worth a shot, though.

Comment (derwolfe) [inline src/cloudpassage_lib/core.clj]: What I'm finding a bit confusing is that if I use raise at 82, it doesn't seem like manifold captures it - it is like it just does a bare raise that clojure handles.

Comment (lvh) [inline src/cloudpassage_lib/core.clj]: That's definitely a bug then; see section starting with "If any stage in chain throws an exception"... at https://github.com/ztellman/manifold/blob/master/docs/deferred.md

Comment (derwolfe) [inline src/cloudpassage_lib/core.clj]: hmm - That could be an artifact of me derefing in the test. That would unwrap the exception and cause it show up. I'll experiment a bit further.

Comment (derwolfe) [inline src/cloudpassage_lib/core.clj]: That was the issue :-). My bug.

**Pull request closed without merging**